### DISCUSSION TITLE
Can I use Git on Github pages

### DISCUSSION THREAD

**Discussion by ghost** [community/community#23947] [New to GitHub]
<p>Can I use Git and host the website on Github pages? Will it run?</p>

**Comment by jfullstackdev** [comment]
<p>Git is a local installation. And whether you use Jekyll, you still need to push files on a GitHub repo, remember, GitHub is web-based, meaning it’s online.</p>
<p>Your question is just like you are asking if the <code>localhost</code> can be accessed through the Internet, it can be but you need to make it available through the larger Internet by port forwarding then securing a domain.</p>

**Comment by ghost** [comment]
<p>Oh ok. I found many working git. So I can use git for my github pages so other stuff can work.</p>

**Comment by jfullstackdev** [comment]
<p>yup, you still push those to a GitHub repo, that GitHub repo will be turned into a site, then you already have your site</p>

**Comment by ghost** [comment]
<p>Oh ok. Thanks for the info. Although I still confused how to paste it to a code</p>

**Comment by jfullstackdev** [comment, marked as answer]
<p>just to give you an idea, Git was created for Linux developers to collaborate, so that housekeeping would not be done manually, just like if there are a hundred developers and each would change a specific file, of course, you as part of the team, you would find it difficult to find their changes,</p>
<p>now what is missing is that, Git is just a revision control, but there was no centralized space so that each would be seeing each others work, from time to time, they would be downloading the codebase and have their changes locally then send the changes to the maintainer/s, then he/she/they would decide whether they would accept such changes,</p>
<p>so GitHub was created so there would be one centralized space and you don’t need to send an e-mail, simply create a pull request,</p>
<p>check this out,</p>
<aside class="onebox githubfolder">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">
      <a href="https://github.com/xdvrx1/what-is-github" target="_blank" rel="noopener nofollow ugc">github.com</a>
  </header>
  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/345;"><img src="https://user-images.githubusercontent.com/47092464/181080359-dd844a5d-d9c4-4cc4-b278-8f2435b9f017.jpeg" class="thumbnail" width="690" height="345" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/9/e/9edafd090e622946642a431eac73b19b745a0277_2_690x345.jpeg, https://aws1.discourse-cdn.com/github/optimized/3X/9/e/9edafd090e622946642a431eac73b19b745a0277_2_1035x517.jpeg 1.5x, https://aws1.discourse-cdn.com/github/original/3X/9/e/9edafd090e622946642a431eac73b19b745a0277.jpeg 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/9/e/9edafd090e622946642a431eac73b19b745a0277_2_10x10.png"></div>

<h3><a href="https://github.com/xdvrx1/what-is-github" target="_blank" rel="noopener nofollow ugc">xdvrx1/what-is-github</a></h3>


  <p><span class="label1">a tech page mentioned GitHub, then you went there but realized you don't have any idea about it or don't have the full understanding, so read this</span></p>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

**Comment by ghost** [comment]
<p>Oh ok. Thanks for the info. I will check it out.</p>
